src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,41 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectTodo: Todo;
+  setSelectTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectTodo, setSelectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasLoading, setHasLoading] = useState(false);
+
+  useEffect(() => {
+    const userData = async () => {
+      setHasLoading(true);
+      try {
+        const hasUserData = await getUser(selectTodo.userId);
```

Comment:
The variable `hasUserData` should be renamed to `userData` or `fetchedUser` to better convey that it holds the user data fetched from the API.